### PR TITLE
Upload key package before publishing identity updates

### PR DESCRIPTION
**Pull request by neekolas**
### TL;DR
Implements parts of https://github.com/xmtp/xmtp-node-go/issues/399

- Uses `UploadKeyPackage` instead of `RegisterInstallation` so we can deprecate the register installation endpoint
- Uploads key package _before_ publishing the identity updates for new installations. This ensures that any installation seen in an identity update must have a matching key package.
- No longer checks the lifetime of key packages when verifying

### What changed?

- Removed expiration checks and set expiration to 0 in MLS validation service
- Updated the client registration process to register identity before applying signature request
- Renamed `register_installation` to `upload_key_package` in API client calls
- Removed lifetime validity checks from key package verification

### CODE REVIEW

**Review by nplasterer — APPROVED** [review]

**Comment by neekolas** [comment]
### Merge activity

* **Aug 14, 5:32 PM PDT**: @neekolas started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/xmtp/libxmtp/962).
